pkg/asset/machines/aws/machines.go

```diff
@@ -247,6 +247,7 @@ func ConfigMasters(machines []machineapi.Machine, controlPlane *machinev1.Contro
 	for _, machine := range machines {
 		providerSpec := machine.Spec.ProviderSpec.Value.Object.(*machineapi.AWSMachineProviderConfig)
 		providerSpec.LoadBalancers = lbrefs
+		providerSpec.PublicIP = pointer.Bool(false)
```

Comment:
Could this break the public-only-subnets case?

Reply:
Yeah: that implementation may depend on this default. That job would need to edit these manifests to set public ip to true.